mlflow/pipelines/regression/v1/pipeline.py

```diff
@@ -185,9 +183,47 @@ class RegressionPipeline(_BasePipeline):
         RegisterStep,
     )
 
-    def _get_step_classes(self) -> List[BaseStep]:
+    _SCORING_DAG_NAME = "scoring_dag"
+    _SCORING_DAG_STEPS = (
+        # Batch scoring DAG
+        IngestScoringStep,
+        PredictStep,
+    )
+
+    _PIPELINE_STEPS = _SCORING_DAG_STEPS + _TRAIN_DAG_STEPS
+
+    _STEPS_SUBGRAPH_MAP = dict()
+    for step_class in _TRAIN_DAG_STEPS:
+        _STEPS_SUBGRAPH_MAP[step_class] = _TRAIN_DAG_NAME
+    for step_class in _SCORING_DAG_STEPS:
+        _STEPS_SUBGRAPH_MAP[step_class] = _SCORING_DAG_NAME
+
+    _SUBGRAPH_INDICES_MAP = {
+        _TRAIN_DAG_NAME: (
+            _PIPELINE_STEPS.index(_TRAIN_DAG_STEPS[0]),
+            _PIPELINE_STEPS.index(_TRAIN_DAG_STEPS[-1]) + 1,
```

Comment:
I'd prefer to not "+1" here to respect what `index` returns for the last step for each DAG, then on L222 return `return self._steps[s:e+1]` to include the last step for each graph.